src/vivarium_census_prl_synth_pop/components/person.py

```diff
@@ -253,25 +185,47 @@ def _perform_gq_person_moves(self, pop: pd.DataFrame, movers: pd.Index) -> pd.Da
 
         return pop
 
-    def _get_new_household_ids(self, pop: pd.DataFrame, sims_who_move: pd.Index) -> pd.Series:
-        households = pop["household_id"]
-        all_household_ids = list(
-            households.squeeze().drop_duplicates()
-        )  # all household_ids in simulation
+    def _perform_non_reference_person_moves(
+        self, pop: pd.DataFrame, movers: pd.Index
+    ) -> pd.DataFrame:
+        """
+        Move each simulant in movers to a random (different) non-GQ household
+        with relationship "Other nonrelative".
+        """
+        if len(movers) == 0:
+            return pop
 
-        new_household_ids = pop.loc[
-            sims_who_move, "household_id"
-        ].copy()  # People who move household_ids
+        non_gq_address_ids = self._get_address_ids_by_household_id(
+            pop[pop["housing_type"] == "Standard"]
+        )
 
+        # NOTE: Unlike in GQ person moves, we require that a "move"
+        # not start and end in the same household.
+        to_move = movers
+        household_id_values = pop.loc[movers, "household_id"]
         additional_seed = 0
-        while (new_household_ids == pop.loc[sims_who_move, "household_id"]).any():
-            unchanged_households = new_household_ids == pop.loc[sims_who_move, "household_id"]
-            new_household_ids[unchanged_households] = self.randomness.choice(
-                new_household_ids.loc[unchanged_households].index,
-                all_household_ids,
-                additional_key=f"newHousehold_ids_{additional_seed}",
+
+        while not to_move.empty:
```

Comment:
Use a `for` loop with a maximum iterations to avoid the potential of an infinite loop.